apps/web/src/pages/integrations/components/multi-provider/SelectPrimaryIntegrationModal.tsx

```diff
@@ -0,0 +1,261 @@
+    max-height: 70vh;
+    padding-right: 5px;
+
+    @media screen and (max-width: 1366px) {
+      max-height: 400px;
+    }
+  }
+`;
+
+const Description = styled(Text)`
+  color: ${colors.B60};
+  font-size: 14px;
+  line-height: 20px;
+`;
+
+const Warning = styled(Text)`
+  color: ${colors.error};
+  font-size: 14px;
+  line-height: 20px;
+`;
+
+const MetaHolder = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 16px;
+`;
+
+const ButtonsHolder = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  align-items: flex-end;
+  gap: 24px;
+  flex: 1 0 0;
+  margin-top: 14px;
+`;
+
+const columns: IExtendedColumn<ITableIntegration>[] = [
+  {
+    accessor: 'name',
+    Header: 'Name',
+    Cell: IntegrationNameCell,
+  },
+  {
+    accessor: 'provider',
+    Header: 'Provider',
+    Cell: withCellLoading(
+      ({ row: { original } }) => {
+        return (
+          <Text data-test-id="integration-provider-cell" rows={1}>
+            {original.provider}
+          </Text>
+        );
+      },
+      { loadingTestId: 'integration-provider-cell-loading' }
+    ),
+  },
+  {
+    accessor: 'active',
+    Header: 'Status',
+    width: 125,
+    maxWidth: 125,
+    Cell: IntegrationStatusCell,
+  },
+];
+
+export interface ISelectPrimaryIntegrationModalProps {
+  isOpened: boolean;
+  environmentId?: string;
+  channelType?: ChannelTypeEnum;
+  onClose: () => void;
+}
+
+export const SelectPrimaryIntegrationModal = ({
+  isOpened,
+  environmentId,
+  channelType,
+  onClose,
+}: ISelectPrimaryIntegrationModalProps) => {
+  const theme = useMantineTheme();
+  const [isPopoverOpened, setPopoverOpened] = useState(false);
+  const [{ selectedIntegrationId, isActive, selectedRowId }, setSelectedState] = useState<{
+    selectedIntegrationId?: string;
+    isActive: boolean;
+    selectedRowId?: string;
+  }>({ isActive: true });
+
+  const { environments, isLoading: areEnvironmentsLoading } = useFetchEnvironments();
+  const environmentName = environments?.find((el) => el._id === environmentId)?.name ?? '';
+
+  const { integrations, loading: areIntegrationsLoading } = useIntegrations();
+  const { makePrimaryIntegration, isLoading: isMarkingPrimaryIntegration } = useMakePrimaryIntegration({
+    onSuccess: onClose,
+  });
+  const integrationsByEnvAndChannel = useMemo<ITableIntegration[]>(() => {
+    const filteredIntegrations = (integrations ?? []).filter((el) => {
+      if (environmentId) {
+        return el.channel === channelType && el._environmentId === environmentId;
+      }
+
+      return el.channel === channelType;
+    });
+
+    return filteredIntegrations.map((el) => mapToTableIntegration(el, environments));
+  }, [integrations, environments, channelType, environmentId]);
+
+  const initialSelectedIndex = useMemo(() => {
+    const primary = findPrimaryIntegration(integrationsByEnvAndChannel ?? []);
+
+    return integrationsByEnvAndChannel?.indexOf(primary);
+  }, [integrationsByEnvAndChannel]);
```

Comment:
find the initial index for a primary integration which will be selected in the modal